src/xGadgetron/pGadgetron/Gadgetron.py

```diff
@@ -1590,3 +1601,45 @@ def calc_rpe_dcw(ad):
     dcw.fill(density_weight)
     
     return dcw
+
+def extract_calibration_data(ad):
+    '''
+    Function to extract all Acquisitions from an AcquisitionData object that
+    carry the coil calibration flag.
+    '''
+
+    if not isinstance(ad, AcquisitionData):
+        raise AssertionError("The trajectory you asked for is not among the available trajectoryies")
+
+    ref_flag = 20
+    ref_img_flag = 21
+
+    subset_idx = numpy.array([])
+    for i in range(ad.number()):
+        acq = ad.acquisition(i)
+        flags = decode_ismrmrd_flag(acq.flags())
+        if ref_flag in flags or ref_img_flag in flags:
+            subset_idx = numpy.append(subset_idx, i)
+
+    if subset_idx.size > 0:
+        calib_data = ad.get_subset(subset_idx)
+        return calib_data
+    else:
+        print("Warning, there are no calibration data in this dataset. We use them all for CSM computation.")
```

Comment:
```suggestion
        print("Warning, there are no calibration data in this dataset. We use them for CSM computation.")
```
but ideally you add something about what the consequences are (if easy?)

Reply:
Done by deletion.